cli/azd/pkg/tools/kubectl/kubectl.go

```diff
@@ -331,6 +336,13 @@ func (cli *kubectlCli) executeCommandWithArgs(
 		args = args.WithCwd(cli.cwd)
 	}
 
+	env := cli.env
+	if cli.kubeConfig != "" {
+		env[KubeConfigEnvVarName] = cli.kubeConfig
```

Comment:
This does end up modifying the `cli.env` map. We could make `SetKubeConfig` modify `env` directly instead to express the same idea more cleanly.

I do feel long-term there should be a more uniform way for us to express the idea in code:
 - For given tool X, environment variables `ENV_FOO`, `ENV_BAR` will be passed through in azd, that doesn't require the more elaborate setup.
I wonder if we feel this is something worth exploring.

Reply:
I like this strategy and updated `SetKubeConfig` to just modify the `cli.env` directly for now.  Will hold off on a larger strategy for a future PR.